51.md

```diff
@@ -6,148 +6,128 @@ Lists
+| Pin list         | 10001 | events the user intends to showcase in their profile page | `"e"` (kind:1 notes)                              |
+| Bookmarks list   | 10003 | things the user intends to save for the future            | `"e"` (kind:1 notes), `"a"` (kind:30023 articles) |
+| Communities list | 10004 | [NIP-72](72.md) communities the user belongs to           | `"a"` (kind:34550 community definitions)          |
 
-## Parameterized Replaceable List Event Example
+## Sets
 
-Lets say a user wants to create a 'Categorized People' list of `nostr` people and has keys:
-```
-priv: fb505c65d4df950f5d28c9e4d285ee12ffaf315deef1fc24e3c7cd1e7e35f2b1
-pub: b1a5c93edcc8d586566fde53a20bdb50049a97b15483cb763854e57016e0fa3d
-```
-The user wants to publicly include these users:
+Sets are lists with well-defined meaning that can enhance the functionality and the UI of clients that rely
+on them. Unlike standard lists, users are expected to have more than one set of each kind, therefore each of
+them must be assigned a different `"d"` identifier.
 
-```json
-["p", "3bf0c63fcb93463407af97a5e5ee64fa883d107ef9e558472c4eb9aaaefa459d"],
-["p", "32e1827635450ebb3c5a7d12c1f8e7b2b514439ac10a67eef3d9fd9c5c68e245"]
-```
-and privately include these users (below is the JSON that would be encrypted and placed in the event content):
+For example, _relay sets_ can be displayed in a dropdown UI to give users the option to switch to which
+relays they will publish an event or from which relays they will read the replies to an event; _curation sets_
+can be used by apps to showcase curations made by others tagged to different topics.
 
-```json
-[
-    ["p", "9ec7a778167afb1d30c4833de9322da0c08ba71a69e1911d5578d3144bb56437"],
-    ["p", "8c0da4862130283ff9e67d889df264177a508974e2feb96de139804ea66d6168"]
-]
-```
+Aside from their main identifier, the `"d"` tag, sets can optionally have a `"title"`, an `"image"` and a
+`"description"` tag that can be used to enhance their UI.
+
+| name          | kind  | description                                                                                   | expected tag items                                |
+| ---           | ---   | ---                                                                                           | ---                                               |
+| Follow sets   | 30000 | categorized groups of users a client may choose to check out in different circumstances       | `"p"` (pubkeys)                                   |
+| Relay sets    | 30002 | user-defined relay groups the user can easily pick and choose from during variadic operations | `"relay"` (relay URLs)                            |
+| Curation sets | 30004 | groups of articles picked by users as interesting and/or belonging to the same category       | `"a"` (kind:30023 articles), `"e"` (kind:1 notes) |
```

Comment:
Suggestion (probably for another day)

```
| Filter sets | 30005 | a list of REQ filters defining custom feeds | `"req"` (a JSON-encoded REQ filter) |
```

Reply:
We're doing something similar in YakiHonne related to the user preferences for content customizations via `kind-30100`. I think its better to include such specifications once for all because soon or later feeds customization will be faced (at least thats what we experienced).

![content_settings](https://github.com/nostr-protocol/nips/assets/29741039/db1b35f3-a884-422b-8252-8f0ca353d85d)
![topics](https://github.com/nostr-protocol/nips/assets/29741039/cd974bee-589f-4d0e-b40a-65a9290447bc)

```
{
  "id": "ed4291ff8ba4343ca444d03624afcee98257aba4af17410a73144ffdc369c467",
  "kind": 30100,
  "pubkey": "1cb0ebecfbe8e291d46c4e082df1e6a183bb321245ef29df5e2703f84b1d86e1",
  "created_at": 1700102857,
  "content": "",
  "tags": [
    [
      "d",
      "MyFavoriteTopicsInYakihonne"
    ],
    [
      "t",
      "YakiHonne"
    ],
    [
      "t",
      "Technology"
    ],
    [
      "t",
      "Crypto"
    ],
    [
      "t",
      "Travel"
    ],
    [
      "t",
      "Social"
    ],
    [
      "t",
      "Nostr"
    ],
    [
      "t",
      "Economy"
    ],
    [
      "t",
      "Finance"
    ],
    [
      "t",
      "Entrepreneurship"
    ],
    [
      "t",
      "Self improvement"
    ],
    [
      "t",
      "Food"
    ],
    [
      "t",
      "Marketing"
    ],
    [
      "t",
      "Art"
    ],
    [
      "t",
      "Photography"
    ],
    [
      "t",
      "Entertainment"
    ],
    [
      "t",
      "Music"
    ]
  ],
  "sig": "4fdc6a423bc7320b432e2edc1e794895468894b92ad7bb9e119bd16c1a6c1d0098bbd8a33ac25856acbae222fda565d7407c17d2b09f6b326de3762ad30a8a1c"
}
```